front-end/src/components/Products.jsx

```diff
@@ -1,11 +1,91 @@
-import React from 'react';
+import React, { useEffect, useState, useContext } from 'react';
+import axios from 'axios';
+import './style/Products.css';
+import { BeerContext } from '../context/context';
 import MenuTop from './MenuTop';
+import { Redirect, Link } from 'react-router-dom';
+import CardProducts from './CardProducts';
+
+const instance = axios.create({ baseURL: 'http://localhost:3001' });
+
+const addTobascket = (price, name, setCart) => {
+  const item = { name, price };
+  setCart(currentState => [...currentState, item]);
+  const cart = localStorage.getItem('cart');
+  if (cart) {
+    localStorage.setItem('cart', JSON.stringify([
+      ...JSON.parse(cart), {
+        name, price,
+      }]));
+  } else {
+    localStorage.setItem('cart', JSON.stringify([{
+      name, price,
+    }]));
+  }
+}
+
+const removeToBascket = (cart, name, setCart) => {
+  if (cart.length === 0) return;
+  const remove = cart.filter(e => e.name !== name);
+  localStorage.setItem('cart', JSON.stringify(remove))
+  setCart(remove);
+}
+
+const renderProducts = (dataApi, cart, setCart) =>
+  dataApi.map(({ id, name, price, urlImage }, index) => {
+    return (
+      <>
+        <div key={id} className="card-products">
+          <CardProducts
+            index={index}
+            name={name}
+            price={price}
+            urlImage={urlImage}
+            classImg="img-products"
+          />
+          <button type="button"
+            onClick={() => addTobascket(price, name, setCart)} data-testid={`${index}-product-plus`}>+</button>
+          <button type="button"
+            onClick={() => removeToBascket(cart, name, setCart)} data-testid={`${index}-product-minus`}>-</button>
+          <button type="button" data-testid={`${index}-product-qtd`} >{cart.length.toString()}</button>
```

Comment:
Aqui na hora de pegar o total do carrinho ele pega o total de itens do carrinho e retorna, por isso que estava aparecendo o mesmo número para todos os itens. Minha sugestão é fazer um filter pelo nome do produto:
```suggestion
        <button type="button" data-testid={`${index}-product-qtd`}>{cart.filter((e) => e.name === name).length}</button>

```